cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -1131,14 +1153,30 @@ impl<I: VCodeInst> MachBuffer<I> {
         self.island_deadline = UNKNOWN_LABEL_OFFSET;
         self.island_worst_case_size = 0;
 
-        // First flush out all constants so we have more labels in case fixups
-        // are applied against these labels.
+        // First flush out all traps/constants so we have more labels in case
+        // fixups are applied against these labels.
         for MachLabelConstant { label, align, data } in mem::take(&mut self.pending_constants) {
             self.align_to(align);
             self.bind_label(label);
             self.put_data(&data[..]);
         }
 
+        for MachLabelTrap {
```

Comment:
How did you decide to put the constant pool before the trap pool? I'm thinking depending on the alignment requirements either order could be best for a given function, so I'm wondering if you had some insight about which was is better or if it just doesn't matter that much.

Reply:
I'd even maybe suggest putting the trap-pool first: it keeps all machine code continuous, which is less surprising both to disassemblers and to humans.

Reply:
Ah yes indeed no reason for it to be second! I like the idea of putting it first too.